proposals/target-typed-new.md

```diff
@@ -0,0 +1,100 @@
+
+# Target-typed `new` expressions
+
+* [x] Proposed
+* [x] [Prototype](https://github.com/alrz/roslyn/tree/features/implicit-new)
+* [ ] Implementation
+* [ ] Specification
+
+## Summary
+[summary]: #summary
+
+Do not require type specification for constructors when the type is known. 
+
+## Motivation
+[motivation]: #motivation
+
+Allow field initialization without duplicating the type.
+```cs
+Dictionary<string, List<int>> field = new() {
+    { "item1", new() { 1, 2, 3 } }
+};
+```
+Allow omitting the type when it can be inferred from usage.
+```cs
+XmlReader.Create(reader, new() { IgnoreWhitespace = true });
+```
+
+## Detailed design
+[design]: #detailed-design
+
+The *object_creation_expression* syntax would be modified to make the *type* optional when parentheses are present. This is required to address the ambiguity with *anonymous_object_creation_expression*.
+```antlr
+object_creation_expression
+    : 'new' type? '(' argument_list? ')' object_or_collection_initializer?
+    | 'new' type object_or_collection_initializer
+    ;
+```
+The target-type is resolved through a conversion that only exists if and only if there exists a single best accessible constructor with regard to provided arguments. Target of the initialization is determined after the conversion succeeded.
+
+The type of the expression would be inferred from the target-type which would be required to be one of the following:
+
+- **Any struct type**
+- **Any reference type**
+- **Any type parameter** with a constructor constraint
+
+The following types do not contribute to the conversion:
```

Comment:
Perhaps say "with the following exceptions", as it otherwise appears to disagree with the above.